### ISSUE TITLE
HTKFeatureDeserializer in python can't read two scps

### ISSUE THREAD

**Issue by staplesinLA** [Microsoft/CNTK#1548]
first, I feed two scps (A-257 and B-257) which both contains 257-dimensional features, it works normally.
then I change A to 257*7-dimensional, it raised error:Floating point exception (core dump).

I checked it by feeding "A-1799 and A-1799" or  “B-257 and B-257", they also worked normally.
How to solve it？
Thanks！

`feature_dim = 1799
feature_dim2 = 257

features_file = "/home/dell/test_py/feature1_normed.scp" 
labels_file = "/home/dell/test_py/feature2_normed.scp" 

fd = HTKFeatureDeserializer(StreamDefs(
    amazing_features = StreamDef(shape=feature_dim, scp=features_file)))#

ld = HTKFeatureDeserializer(StreamDefs(
    awesome_labels = StreamDef(shape=feature_dim2, scp=labels_file)))#

reader = MinibatchSource([fd,ld]) `

**Comment by eldakms** [comment]
hi staplesinLA,
I'm trying to reproduce this on the test data (<CNTK>/Examples/Speech/AN4/Data) but am still unsuccessful. Could you please share the full repro if possible? (either with data or maybe on AN4 data?)

Where does to core dump happen? when the first minibatch is requested from the minibatch source? or when the data is feed into the network?
Thanks!

**Comment by eldakms** [comment]
I am doing something like that, but cannot reproduce the crash:

    feature_dim = 33
    feature_dim2 = 165

    fd = HTKFeatureDeserializer(StreamDefs(
        amazing_features = StreamDef(shape=feature_dim, scp=features_file)))

    fd1 = HTKFeatureDeserializer(StreamDefs(
        amazing_features1 = StreamDef(shape=feature_dim2, scp=features_file)))

    reader = MinibatchSource([fd, fd1])

**Comment by staplesinLA** [comment]
Thanks for your reply @eldakms ! Here is my complete script:
`# -*- coding:UTF-8 -*-
import cntk as C
import numpy as np
from cntk.io import MinibatchSource, HTKFeatureDeserializer, HTKMLFDeserializer, StreamDef, StreamDefs
from cntk.blocks import LSTM, Placeholder, Input
from cntk.layers import Recurrence, Dense, BatchNormalization
from cntk.models import Sequential, For
import os, sys

mbsize = 256
lr = [0.001] 

feature_dim = 257
feature_dim2 = 257
features_file = "/home/dell/test_py/first_feature.scp" 
labels_file = "/home/dell/test_py/second_feature.scp"


fd = HTKFeatureDeserializer(StreamDefs(
    amazing_features = StreamDef(shape=feature_dim, scp=features_file)))

ld = HTKFeatureDeserializer(StreamDefs(
    awesome_labels = StreamDef(shape=feature_dim, scp=labels_file)))

reader = MinibatchSource([fd,ld])


features = C.input_variable((feature_dim)) 
labels = C.input_variable((feature_dim2)) 
input_map={ features: reader.streams.amazing_features, labels: reader.streams.awesome_labels }

	
model = Sequential ([
    For(range(3), lambda: \
        Dense(2048, activation=C.sigmoid)),
    Dense(257, activation=C.sigmoid)
])	
					

z = model(features) 
se = C.squared_error(z, labels) 

learner = C.sgd(z.parameters, 
                    lr=C.learning_rate_schedule(lr, C.UnitType.sample))
    
trainer = C.Trainer(z, (se,None), learner)


num_minibatches=1000
print_frequency_mbsize = num_minibatches // 50
pp = C.ProgressPrinter(print_frequency_mbsize)
for i in range(num_minibatches):
    mb_data = reader.next_minibatch(mbsize, input_map=input_map)
    trainer.train_minibatch(mb_data)
    pp.update_with_trainer(trainer)
    trainer_loss = C.utils.get_train_loss(trainer)
	

z.save('model.dnn')`

**Comment by staplesinLA** [comment]
It seems the crash happen at the beginning.
————————————————————————————————————
>>>python test.py 
Reading script file /home/dell/test_py/noisy_normed.scp ... 11550 entries
HTKDataDeserializer::HTKDataDeserializer: selected 11550 utterances grouped into 25 chunks, average chunk size: 462.0 utterances, 88097.7 frames (for I/O: 462.0 utterances, 88097.7 frames)
HTKDataDeserializer::HTKDataDeserializer: determined feature kind as 257-dimensional 'USER' with frame shift 16.0 ms
Reading script file /home/dell/test_py/clean_normed.scp ... 11550 entries
HTKDataDeserializer::HTKDataDeserializer: selected 11550 utterances grouped into 25 chunks, average chunk size: 462.0 utterances, 88097.7 frames (for I/O: 462.0 utterances, 88097.7 frames)
HTKDataDeserializer::HTKDataDeserializer: determined feature kind as 257-dimensional 'USER' with frame shift 16.0 ms
-------------------------------------------------------------------
Build info: 

		Built time: Feb 22 2017 13:29:08
		Last modified date: Sun Feb 19 16:12:09 2017
		Build type: release
		Build target: GPU
		With 1bit-SGD: yes
		With ASGD: yes
		Math lib: mkl
		CUDA_PATH: /usr/local/cuda-8.0
		CUB_PATH: /usr/local/cub-1.4.1
		CUDNN_PATH: /usr/local/cudnn-5.1
		Build Branch: HEAD
		Build SHA1: bfd2a348763305e37eca733cf9a659d22514eefd
		Built by Source/CNTK/buildinfo.h$$0 on a5a5e4cc79e0
		Build Path: /home/philly/jenkins/workspace/CNTK-Build-Linux
		MPI distribution: Open MPI
		MPI version: 1.10.3
-------------------------------------------------------------------
Floating point exception (core dump)

**Comment by staplesinLA** [comment]
What‘s strange is when i change those two feature files above to two same files , the crash disappeared.
Like, 
`features_file = "/home/dell/test_py/first_feature.scp"
labels_file = "/home/dell/test_py/first_feature.scp"`
or
`features_file = "/home/dell/test_py/second_feature.scp"
labels_file = "/home/dell/test_py/second_feature.scp"`.

**Comment by eldakms** [comment]
Will it be possible to share the scp + data files with me? Thanks!

**Comment by eldakms** [comment]
you can drop an email directly - eldak at microsoft dot com

**Comment by staplesinLA** [comment]
@eldakms ok, don't know why i didn't receive your reply from github push.
I'll immediately prepare a pack to you!
Much thanks.

**Comment by eldakms** [comment]
Thanks! I would appreciate if you could send it by email.

**Comment by eldakms** [comment]
Thank you very much for sending me the repro!

I will add a better error message for this, but your scp data is not correct.
If you combine two deserializers, you need to **make sure the logical keys of the sequences match**.

If you look at two files (clean and noisy), then we cannot match them because they have different logical keys.

Noisy scp:
S1_noisyNormed.lsp=path to noisy features

Clean scp:
S1_cleanNormed.lsp=path to clean features

If I change scps so that the logical keys match:

Noisy:
S1.lsp=path to noisy features

Clean:
S1.lsp=path to clean features

Then I can successfully train with your data. I'm closing this, feel free to submit other issues if you bump into anything. Thanks!
